flight_sql/accessors/timestamp_array_accessor.cc

```diff
@@ -32,23 +34,32 @@ int64_t GetConversionToSecondsDivisor(TimeUnit::type unit) {
   return divisor;
 }
 
-uint32_t CalculateFraction(TimeUnit::type unit, uint64_t units_since_epoch) {
+uint32_t CalculateFraction(TimeUnit::type unit, int64_t units_since_epoch) {
   // Convert the given remainder and time unit to nanoseconds
   // since the fraction field on TIMESTAMP_STRUCT is in nanoseconds.
   switch (unit) {
   case TimeUnit::SECOND:
     return 0;
   case TimeUnit::MILLI:
     // 1000000 nanoseconds = 1 millisecond.
+    if (units_since_epoch < 0)
+      units_since_epoch += driver::odbcabstraction::MILLI_TO_SECONDS_DIVISOR *
+          (1 + std::abs(units_since_epoch / driver::odbcabstraction::MILLI_TO_SECONDS_DIVISOR));
```

Comment:
For this expression we already know that std::abs() will negate the value to get a positive since we've already checked that units_since_epoch < 0. Would it be better to just negate? Negation is usually its own instruction.

Reply:
Abs should be its own instruction too; I think abs is clearer as to the intention?

I /should/ probably comment (so people don't have to check the Standard) that it's going to round towards zero.

Reply:
What I should do though is fix this equivalent to https://github.com/dremio/flightsql-odbc/pull/3#pullrequestreview-2012515213 (below comment about explanation) so it doesn't have the undesirable case where it adds too much (and ends up taking modulo of the (positive) divisor by itself...).

Reply:
> What I should do though is fix this equivalent to [#3 (review)](https://github.com/dremio/flightsql-odbc/pull/3#pullrequestreview-2012515213) (below comment about explanation) so it doesn't have the undesirable case where it adds too much (and ends up taking modulo of the (positive) divisor by itself...).

Did this, and commented wrt rounding direction.